packages/sdk-cli/src/api/baseAPI.ts

```diff
@@ -1,7 +1,41 @@
+import * as t from 'io-ts';
+import { decode } from '@fitbit/jsonrpc-ts';
+
 import * as auth from '../auth';
 import environment from '../auth/environment';
 import fetch, { Headers } from '../fetch';
-import { decodeJSON } from '../util/fetchUtil';
+import { assertJSON, okOrElse } from '../util/fetchUtil';
+
+// tslint:disable-next-line:variable-name
+const APIErrorResponse = t.type(
+  {
+    errors: t.array(t.type({
+      message: t.string,
+    })),
+  },
+  'Hosts403Response',
+);
+type APIErrorResponse = t.TypeOf<typeof APIErrorResponse>;
+
+export const assertAPIResponseOK = okOrElse((response) => {
+  return assertJSON()(response)
+    .catch(() => Promise.reject(
+      // tslint:disable-next-line:max-line-length
+      new Error(`Fetch of ${response.url} returned status ${response.status} ${response.statusText}`),
+    ))
+    .then(responseObject => Promise.reject(new Error(APIErrorResponse.decode(responseObject).fold(
+      // tslint:disable-next-line:max-line-length
+      () => `fetch of ${response.url} returned status ${response.status} ${response.statusText} with body: ${JSON.stringify(responseObject, undefined, 2)}`,
+      errorObj => errorObj.errors.map(err => err.message).join('\n'),
+    )))
+    .then(() => Promise.resolve(response)));
```

Comment:
This is a bit of a hack, it'll never get here as we see that it rejects above. But typescript complains that the function has return type `Promise<never>` and I spent enough time bashing my head trying to get it to work. Open to suggestions on working around this.